code/modules/mob/living/simple_animal/hostile/giant_spider/effects.dm

```diff
@@ -44,6 +45,23 @@
 	health -= damage
 	healthcheck()
 
+/obj/effect/spider/attack_hand(var/mob/living/carbon/human/user)
+	if(user.a_intent == I_HURT)
+		user.delayNextAttack(8)
+		user.visible_message("<span class='danger'>[user.name] claws at \the [src]!</span>", \
+			"<span class='danger'>You claw at \the [src]!</span>", \
+			"You hear something tear.")
+		health -= 2
+		healthcheck()
+	else
+		var/atom/movable/I = pick(contents)
```

Comment:
If they are empty, there's something else that's wrong.  I'd argue that a runtime here would serve the purpose of an assert.